File: src/console/command/parameter/def/StringParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\CommandParameter;

readonly class StringParameter extends CommandParameter {

    public function __construct(
        string $name,
        bool $optional,
        private bool $multiString = false
    ) {
        parent::__construct($name, $optional);
    }

    public function parseValue(string $input): string {
        return $input;
    }

    public function getType(): string {
        return "string";
    }

    public function isMultiString(): bool {
        return $this->multiString;
    }
}

File: src/main/java/de/pocketcloud/cloud/console/command/parameter/def/StringParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;
import lombok.Getter;

@Getter
public class StringParameter extends BaseCommandParameter {

    private final boolean multiString;

    public StringParameter(String name, boolean optional, boolean multiString) {
        super(name, optional);
        this.multiString = multiString;
    }

    public StringParameter(String name, boolean optional) {
        this(name, optional, false);
    }

    @Override
    public String parseValue(String input) {
        return input;
    }

    @Override
    public String getType() {
        return "string";
    }
}